rfcs/0001-pip-rearchitecture.md

```diff
@@ -0,0 +1,64 @@
+# Pip Rearchitecture
+
+## Proposal
+
+The existing pip buildpack should be rewritten and restructured to *only*
+provide the `pip` dependency. The `pip install` logic should be factored out
+into it's own buildpack.
+
+## Motivation
+
+In keeping with the overarching [Python Buildpack Rearchitecture
+RFC](https://github.com/paketo-community/python/blob/main/rfcs/0001-restructure.md),
+the Pip Buildpack should perform one task, which is installing the `pip`
+dependency. This is part of the effort in Paketo Buildpacks to reduce the
+responsibilities of each buildpack to make them easier to understand
+and maintain.
+
+## Implementation
+
+The implementation details are outlined in [this
+issue](https://github.com/paketo-community/pip/issues/82). Specifically, the
+new Pip Buildpack will always `detect` and  will always `provide` `pip`. It
+will be the responsibility of a downstream buildpack (such as a future Pip
+Install buildpack) to `require` the `pip` dependency.
+
+The new `provides`/`requires` contract will initially be:
+
+* `pip`
+  * provides: `pip`
+  * requires: `cpython` OR {`python` + `requirements`} during `build`
+
+The {`python` + `requirements`} requirement is included for
+backwards-compatibility and will be removed towards the end of the full Python
+rearchitecture.
+
+
+The final `provides`/`requires` contract will be:
+
+* `pip`
+  * provides: `pip`
+  * requires: `cpython` during `build`
+
+### Configuration
+
+Users may specify a Pip version through the `BP_PIP_VERSION` environment
+variable. This can be set explicitly at build time (e.g. `pack --env`) or through
+`project.toml`.
+
+### Dependency Installation
+
+`Pip` installation involves a few steps:
+
+The buildpack runs `PYTHONUSERBASE=<path/to/pip/layer> python -m pip
+install <path/to/tgz> --user` to install the requested version. Setting the
+`PYTHONUSERBASE` variable ensures that pip is installed to the newly created
+layer.
+
+The pip installation is then added to the `PATH` variable so that it may be
+invoked without the `python -m` prefix.
+
+The final step is exporting `PYTHONPATH=<path-to-pip-layer>/lib/python/site-packages` so that Python
```

Comment:
1. What does "exporting" mean in this context? Setting the env value in the build phase? Or is it writing the env variable to build-time/launch-time env vars in layer for downstream consumers?
2. If it is writing to the layer, maybe it should prepend/append to the existing value?

Reply:
1. We meant the latter, writing the env to the build-time/shared env vars.
2. Yes, we should've called that out. We discussed this and decided it made sense for us to prepend to PYTHONPATH.